src/Processors/Formats/Impl/ArrowColumnToCHColumn.cpp

```diff
@@ -558,10 +567,23 @@ void ArrowColumnToCHColumn::arrowColumnsToCHChunk(Chunk & res, NameToColumnPtr &
                 read_from_nested = nested_tables[nested_table_name]->has(header_column.name);
             }
 
-
-            // TODO: What if some columns were not presented? Insert NULLs? What if a column is not nullable?
             if (!read_from_nested)
-                throw Exception{ErrorCodes::THERE_IS_NO_COLUMN, "Column '{}' is not presented in input data.", header_column.name};
+            {
+                missing_column_indexes.push_back(column_i);
+                if (defaults_for_omitted_fields)
```

Comment:
This is a slightly incorrect use of this setting. If  `defaults_for_omitted_fields` is true, we should evaluate and insert default column expression, if it's `false`, we should just insert column type defaults. So, we should use `defaults_for_omitted_fields` to determine if we need to add info about missing columns to `block_missing_values` (because it's used for evaluating default expressions). Here we need to create and use new settings for each format Arrow/ORC/Parquet (like `input_format_parquet_allow_missing_columns`).

Reply:
And maybe it would be better to remove filling and returning missing_column_indexes from ArrowColumnToCHColumn, because we can calculate it only once in prepareReader in Arrow/ORC/Parquet formats bt arrow schema.

Reply:
Here is my understanding of your comment, am I right? 
- if input_format_parquet_allow_missing_columns is false, then ArrowColumnToCHColumn throw an exception if we can't find columns in `arrow::table`
- if input_format_parquet_allow_missing_columns is true, defaults_for_omitted_fields is true, then we fill missing columns with zero values of its type in case those columns's default expression not defined(`AddingDefaultsTransform` only recalculate default value for columns which have default expression). And Then return missing_column_indexes.
- if input_format_parquet_allow_missing_columns is true, defaults_for_omitted_fields is false, then fill missing columns with zero values, and return empty missing_column_indexes.

Reply:
Yes, you are right

Reply:
I think treat function `getMissingColumns` as non-static is better, because `header`, `import_nested` is already members of `ArrowColumnToCHColumn`

Reply:
done